posts/tests/mock.py

```diff
@@ -16,4 +27,28 @@ def setUpTestData(cls):
     cls.test_post = Post.objects.get(id=response.data["id"])
     
   def tearDown(self):
-    self.writer.object.delete()
\ No newline at end of file
+    self.writer.object.delete()
+    
+
+class PostTestMultiple(TestCase):
+  writers = [WRITER_USER, WRITER_USER2]
+  
+  @classmethod
+  def setUpTestData(cls):
+    for index, writer in enumerate(cls.writers):
+      setattr(cls, f'writer{index}', TestUserSet(**writer))
+      writer = getattr(cls, f'writer{index}')
+      posts_body = [
+        post_body_factory(f"finish and public True Post-{index}", writer.object.name, True, True),
+        post_body_factory(f"only finish True Post-{index}", writer.object.name, True),
+        post_body_factory(f"only public True Post-{index}", writer.object.name, False, True),
+        post_body_factory(f"finish and public False Post-{index}", writer.object.name),
+      ]
```

Comment:
Post모델에서 상태 필드인(Boolean) is_finish, is_public으로 표현할 수 있는 경우의 수가 하드코딩 되어있음.
만약 is_finish, is_public외에 더 많은 상태 필드가 존재하면 코드가 굉장히 걸어짐
Python의 itertools모듈을 사용하면 더 좋은 해결책이 존재.
```suggestion
      posts_body = [dict(
      zip(fields, values), **{"title": "title", "author": "author"}
      ) for values in product([True,False], repeat=len(fields))]
```
title, author부분에만 따로 정의한 포맷형태를 따르면 됨.

Reply:
```
class ModelTestTools:
  state_fields = []
  
  @classmethod
  def all_state_field_cases(cls):
    """
    Generates all combinations of Boolean fields.
    Returns the dictionaries in which the field and value are matched.
    """
    state_field_cases = product([True,False], repeat=len(cls.state_fields))
    cases_to_dicts = [dict(zip(cls.state_fields, values)) for values in state_field_cases]
    return cases_to_dicts
```
state_field로 정의된 필드들이 Boolean이라는 가정하에 만들어낼 수 있는 모든 조합을 생성하는 all_state_field_cases메서드 생성 
```
  @classmethod
  def setUpTestData(cls):
    """
    Generates a post with all combinations of state fields, which are the Boolean fields of the Post model.
    """
    for index, writer in enumerate(cls.writers):
      ...
      state_field_cases = cls.all_state_field_cases()
      posts_body = [
        dict(**case, **{
        "title": f"case-post-{id} : {writer.object.name}",
        "author": writer.object.name,
        }) for id, case in enumerate(state_field_cases)
      ]
      ...
```
is_finish, is_public의 모든 조합을 생성한 state_field_cases를 통해 해당 값을 반복하여 상태필드로 표현할 수 있는 모든 경우의 수 의 포스트 body들을 생성